version: '3'

services:
    db:
        image: postgres:14.4
        volumes:
            - postgres_data:/var/lib/postgresql/data/
        ports:
            - "5432:5432"
        environment:
            - POSTGRES_DB=django
            - POSTGRES_USER=django
            - POSTGRES_PASSWORD=django
        healthcheck:
            test: ["CMD-SHELL", "pg_isready -U django"]
            interval: 10s
            timeout: 5s
            retries: 5

    web:
        container_name: djangoTemplate
        build:
            context: .
            dockerfile: Dockerfile
        environment:
            - DB_NAME=django
            - DB_USER=django
            - DB_PASS=django
            - DB_HOST=db
            - REDIS_HOST=redis
            - REDIS_PORT=6379
        volumes:
            - .:/app
        ports:
            - "8000:8000"
        depends_on:
            db:
                condition: service_healthy
            redis:
                condition: service_started
        restart: on-failure:1
        command: "python manage.py runserver 0.0.0.0:8000"
        env_file:
            -   .env

    sass:
        build:
            context: .
            dockerfile: source_scss/Dockerfile
        volumes:
            - .:/app
        command: "npx sass source_scss:static --watch --no-source-map --style=compressed"

    redis:
        image: "redis:latest"
        ports:
            - "6379:6379"

    celery:
        build:
            context: .
            dockerfile: Dockerfile
        command: "celery -A djangoTemplate worker --loglevel=info"
        volumes:
            - .:/app
        depends_on:
            - db
            - redis
        environment:
            - REDIS_HOST=redis
            - REDIS_PORT=6379
        env_file:
            - .env

volumes:
    postgres_data:
